### PR TITLE
IRQ performance improvement, add CSO exception, more ANSI support

### PR DESCRIPTION
**Pull request by Ticki**

### CODE REVIEW

File: kernel/main.rs [hunk 1]
```diff
@@ -131,29 +131,23 @@ static PIT_DURATION: Duration = Duration {
...
-                }
+        for i in env().contexts.lock().iter() {
+            if i.interrupted {
+                halt = false;
+                break;
             }
         }
 
+        asm!("sti" : : : : "intel", "volatile");
+
```
Comment (jackpot51) [inline kernel/main.rs]: To make this not halt after an interrupt, hlt needs to be the instruction immediately after sti

Comment (Ticki) [inline kernel/main.rs]: Woosh. Will fix immediately.

File: kernel/main.rs [hunk 2]
```diff
@@ -131,24 +131,20 @@ static PIT_DURATION: Duration = Duration {
...
-        {
-            let contexts = ::env().contexts.lock();
-            for i in 1..contexts.len() {
-                if let Some(context) = contexts.get(i) {
-                    if context.interrupted {
-                        halt = false;
-                        break;
-                    }
-                }
+        for i in env().contexts.lock().iter() {
```
Comment (jackpot51) [inline kernel/main.rs]: Context 0 is this function, use .skip(1) on the iter or you will not halt

Comment (Ticki) [inline kernel/main.rs]: Really. I thought it wouldn't iterate over context 0.

**Pull request merged**